diff -up dpdk-stable-18.11.2/mk/rte.sdkdoc.mk.multilib dpdk-stable-18.11.2/mk/rte.sdkdoc.mk
--- dpdk-stable-18.11.2/mk/rte.sdkdoc.mk.multilib	2019-09-11 12:01:46.491478717 +0200
+++ dpdk-stable-18.11.2/mk/rte.sdkdoc.mk	2019-09-11 12:49:19.100816161 +0200
@@ -48,7 +48,7 @@ api-html: $(API_EXAMPLES)
 	         -e "s|@OUTPUT@|$(RTE_OUTPUT)/doc|"           \
 	         -e "s|@HTML_OUTPUT@|html/api|"               \
 	         -e "s|@TOPDIR@|./|g"                         \
-	         -e "s|@STRIP_FROM_PATH@|./|g"                \
+	         -e "s|@STRIP_FROM_PATH@|$(RTE_SDK) $(RTE_OUTPUT)/doc/html/|"                \
 	         $(RTE_SDK)/doc/api/doxy-api.conf.in)|        \
 	    doxygen -
 	$(Q)$(RTE_SDK)/doc/api/doxy-html-custom.sh $(RTE_OUTPUT)/doc/html/api/doxygen.css
