diff -up ./examples/ip_pipeline/pipeline/pipeline_common_fe.c.pipe_common ./examples/ip_pipeline/pipeline/pipeline_common_fe.c
--- ./examples/ip_pipeline/pipeline/pipeline_common_fe.c.pipe_common	2016-02-15 16:51:32.161025145 -0500
+++ ./examples/ip_pipeline/pipeline/pipeline_common_fe.c	2016-02-15 16:55:05.654052482 -0500
@@ -337,7 +337,7 @@ app_link_config(struct app_params *app,
 		return -1;
 	}
 
-	netmask = (~0) << (32 - depth);
+	netmask = UINT_MAX << (32 - depth);
 	host = ip & netmask;
 	bcast = host | (~netmask);
 
@@ -889,7 +889,7 @@ print_link_info(struct app_link_params *
 {
 	struct rte_eth_stats stats;
 	struct ether_addr *mac_addr;
-	uint32_t netmask = (~0) << (32 - p->depth);
+	uint32_t netmask = UINT_MAX << (32 - p->depth);
 	uint32_t host = p->ip & netmask;
 	uint32_t bcast = host | (~netmask);
 
