diff -up dpdk-2.1.0-2898.gitf1e779ec/examples/Makefile.examples dpdk-2.1.0-2898.gitf1e779ec/examples/Makefile
index cee95cd..aa760d2 100644
--- dpdk-2.1.0-2898.gitf1e779ec/examples/Makefile.examples	2015-10-22 22:36:41.000000000 +0300
+++ dpdk-2.1.0-2898.gitf1e779ec/examples/Makefile	2015-10-22 23:36:41.000000000 +0300
@@ -45,7 +45,7 @@ DIRS-y += dpdk_qat
 endif
 DIRS-y += exception_path
 DIRS-y += helloworld
-DIRS-y += ip_pipeline
+#DIRS-y += ip_pipeline
 DIRS-y += ip_reassembly
 DIRS-$(CONFIG_RTE_IP_FRAG) += ip_fragmentation
 DIRS-y += ipv4_multicast
@@ -73,6 +73,6 @@ DIRS-$(CONFIG_RTE_LIBRTE_VHOST) += vhost
 DIRS-$(CONFIG_RTE_LIBRTE_XEN_DOM0) += vhost_xen
 DIRS-y += vmdq
 DIRS-y += vmdq_dcb
-DIRS-$(CONFIG_RTE_LIBRTE_POWER) += vm_power_manager
+#DIRS-$(CONFIG_RTE_LIBRTE_POWER) += vm_power_manager
 
 include $(RTE_SDK)/mk/rte.extsubdir.mk
