diff -up ./examples/ip_pipeline/pipeline/pipeline_firewall.c.pfire ./examples/ip_pipeline/pipeline/pipeline_firewall.c
--- ./examples/ip_pipeline/pipeline/pipeline_firewall.c.pfire	2016-02-16 10:16:03.856528168 -0500
+++ ./examples/ip_pipeline/pipeline/pipeline_firewall.c	2016-02-16 10:17:31.128996838 -0500
@@ -256,10 +256,10 @@ app_pipeline_firewall_key_check_and_norm
 			return -1;
 
 		if (src_ip_depth)
-			src_ip_netmask = (~0) << (32 - src_ip_depth);
+			src_ip_netmask = UINT_MAX << (32 - src_ip_depth);
 
 		if (dst_ip_depth)
-			dst_ip_netmask = ((~0) << (32 - dst_ip_depth));
+			dst_ip_netmask = (UINT_MAX << (32 - dst_ip_depth));
 
 		key->key.ipv4_5tuple.src_ip &= src_ip_netmask;
 		key->key.ipv4_5tuple.dst_ip &= dst_ip_netmask;
