diff -up ./examples/ip_pipeline/pipeline/pipeline_routing.c.pipe_route ./examples/ip_pipeline/pipeline/pipeline_routing.c
--- ./examples/ip_pipeline/pipeline/pipeline_routing.c.pipe_route	2016-02-15 16:37:25.909132363 -0500
+++ ./examples/ip_pipeline/pipeline/pipeline_routing.c	2016-02-15 16:38:28.644903861 -0500
@@ -319,7 +319,7 @@ app_pipeline_routing_add_route(struct ap
 		if ((depth == 0) || (depth > 32))
 			return -1;
 
-		netmask = (~0) << (32 - depth);
+		netmask = UINT_MAX << (32 - depth);
 		key->key.ipv4.ip &= netmask;
 
 		/* data */
@@ -421,7 +421,7 @@ app_pipeline_routing_delete_route(struct
 		if ((depth == 0) || (depth > 32))
 			return -1;
 
-		netmask = (~0) << (32 - depth);
+		netmask = UINT_MAX << (32 - depth);
 		key->key.ipv4.ip &= netmask;
 	}
 	break;
