diff -up ./drivers/net/ixgbe/ixgbe_rxtx.c.ixgbe ./drivers/net/ixgbe/ixgbe_rxtx.c
--- ./drivers/net/ixgbe/ixgbe_rxtx.c.ixgbe	2016-02-15 15:07:07.874139300 -0500
+++ ./drivers/net/ixgbe/ixgbe_rxtx.c	2016-02-15 15:09:18.420830519 -0500
@@ -1574,6 +1574,8 @@ ixgbe_recv_pkts_lro(void *rx_queue, stru
 		uint32_t staterr;
 
 next_desc:
+		next_rxe = NULL;
+
 		/*
 		 * The code in this whole file uses the volatile pointer to
 		 * ensure the read ordering of the status and the rest of the
