diff -up ./drivers/net/bnx2x/bnx2x.h.bnx2x ./drivers/net/bnx2x/bnx2x.h
--- ./drivers/net/bnx2x/bnx2x.h.bnx2x	2016-02-15 15:05:01.900577388 -0500
+++ ./drivers/net/bnx2x/bnx2x.h	2016-02-15 15:06:26.037956379 -0500
@@ -1608,7 +1608,7 @@ void bnx2x_reg_write32(struct bnx2x_soft
 #define INIT_DMAE_C(sc)     ((SC_PORT(sc) * MAX_DMAE_C_PER_PORT) + SC_VN(sc))
 #define PMF_DMAE_C(sc)      ((SC_PORT(sc) * MAX_DMAE_C_PER_PORT) + E1HVN_MAX)
 
-static const uint32_t dmae_reg_go_c[] = {
+static const uint32_t dmae_reg_go_c[] __attribute__((unused)) = {
     DMAE_REG_GO_C0,  DMAE_REG_GO_C1,  DMAE_REG_GO_C2,  DMAE_REG_GO_C3,
     DMAE_REG_GO_C4,  DMAE_REG_GO_C5,  DMAE_REG_GO_C6,  DMAE_REG_GO_C7,
     DMAE_REG_GO_C8,  DMAE_REG_GO_C9,  DMAE_REG_GO_C10, DMAE_REG_GO_C11,
