diff -up dpdk-2.2.0/examples/ip_pipeline/Makefile.examples dpdk-2.2.0/examples/ip_pipeline/Makefile
--- dpdk-2.2.0/examples/ip_pipeline/Makefile.examples	2016-01-20 14:30:19.266011981 +0200
+++ dpdk-2.2.0/examples/ip_pipeline/Makefile	2016-01-20 14:30:28.638006264 +0200
@@ -73,7 +73,7 @@ SRCS-$(CONFIG_RTE_LIBRTE_PIPELINE) += pi
 SRCS-$(CONFIG_RTE_LIBRTE_PIPELINE) += pipeline_routing.c
 
 CFLAGS += -I$(SRCDIR) -I$(SRCDIR)/pipeline
-CFLAGS += -O3
+CFLAGS += -O3 -msse4
 CFLAGS += $(WERROR_FLAGS) -Wno-error=unused-function -Wno-error=unused-variable
 
 include $(RTE_SDK)/mk/rte.extapp.mk
