diff -up ./examples/ip_pipeline/cpu_core_map.c.lengthfix ./examples/ip_pipeline/cpu_core_map.c
--- ./examples/ip_pipeline/cpu_core_map.c.lengthfix	2017-02-14 16:17:45.000000000 -0500
+++ ./examples/ip_pipeline/cpu_core_map.c	2017-02-21 10:23:38.982228068 -0500
@@ -311,7 +311,7 @@ cpu_core_map_get_core_id_linux(int lcore
 static int
 cpu_core_map_get_socket_id_linux(int lcore_id)
 {
-	char buffer[64];
+	char buffer[128];
 	FILE *fd;
 	int socket_id;
 
diff -up ./examples/multi_process/client_server_mp/mp_server/main.c.lengthfix ./examples/multi_process/client_server_mp/mp_server/main.c
--- ./examples/multi_process/client_server_mp/mp_server/main.c.lengthfix	2017-02-14 16:17:45.000000000 -0500
+++ ./examples/multi_process/client_server_mp/mp_server/main.c	2017-02-21 11:17:15.946342832 -0500
@@ -93,14 +93,14 @@ static const char *
 get_printable_mac_addr(uint8_t port)
 {
 	static const char err_address[] = "00:00:00:00:00:00";
-	static char addresses[RTE_MAX_ETHPORTS][sizeof(err_address)];
+	static char addresses[RTE_MAX_ETHPORTS][sizeof(err_address)+2];
 
 	if (unlikely(port >= RTE_MAX_ETHPORTS))
 		return err_address;
 	if (unlikely(addresses[port][0]=='\0')){
 		struct ether_addr mac;
 		rte_eth_macaddr_get(port, &mac);
-		snprintf(addresses[port], sizeof(addresses[port]),
+		snprintf(addresses[port], sizeof(addresses[port])+2,
 				"%02x:%02x:%02x:%02x:%02x:%02x\n",
 				mac.addr_bytes[0], mac.addr_bytes[1], mac.addr_bytes[2],
 				mac.addr_bytes[3], mac.addr_bytes[4], mac.addr_bytes[5]);
diff -up ./examples/server_node_efd/server/main.c.lengthfix ./examples/server_node_efd/server/main.c
--- ./examples/server_node_efd/server/main.c.lengthfix	2017-02-21 10:59:27.769936326 -0500
+++ ./examples/server_node_efd/server/main.c	2017-02-21 10:59:43.719823270 -0500
@@ -100,7 +100,7 @@ static const char *
 get_printable_mac_addr(uint8_t port)
 {
 	static const char err_address[] = "00:00:00:00:00:00";
-	static char addresses[RTE_MAX_ETHPORTS][sizeof(err_address)];
+	static char addresses[RTE_MAX_ETHPORTS][sizeof(err_address)+2];
 	struct ether_addr mac;
 
 	if (unlikely(port >= RTE_MAX_ETHPORTS))
diff -up ./examples/server_node_efd/shared/common.h.lengthfix ./examples/server_node_efd/shared/common.h
--- ./examples/server_node_efd/shared/common.h.lengthfix	2017-02-21 10:48:32.788582926 -0500
+++ ./examples/server_node_efd/shared/common.h	2017-02-21 10:48:07.540762374 -0500
@@ -86,9 +86,9 @@ get_rx_queue_name(unsigned int id)
 {
 	/*
 	 * Buffer for return value. Size calculated by %u being replaced
-	 * by maximum 3 digits (plus an extra byte for safety)
+	 * by maximum 10 digits (plus an extra byte for safety)
 	 */
-	static char buffer[sizeof(MP_NODE_RXQ_NAME) + 2];
+	static char buffer[sizeof(MP_NODE_RXQ_NAME) + 12];
 
 	snprintf(buffer, sizeof(buffer) - 1, MP_NODE_RXQ_NAME, id);
 	return buffer;
