diff -up ./examples/ethtool/ethtool-app/Makefile.makefix ./examples/ethtool/ethtool-app/Makefile
--- ./examples/ethtool/ethtool-app/Makefile.makefix	2017-02-22 07:08:22.873372009 -0500
+++ ./examples/ethtool/ethtool-app/Makefile	2017-02-22 07:09:10.138033792 -0500
@@ -49,6 +49,8 @@ CFLAGS += $(WERROR_FLAGS)
 
 LDLIBS += -L$(subst ethtool-app,lib,$(RTE_OUTPUT))/lib
 LDLIBS += -lrte_ethtool
-
+ifeq ($(CONFIG_RTE_LIBRTE_IXGBE_PMD),y)
+LDLIBS += -lrte_pmd_ixgbe
+endif
 
 include $(RTE_SDK)/mk/rte.extapp.mk
