diff -up ./dpdk-2.1.0/drivers/net/null/rte_eth_null.c.null ./dpdk-2.1.0/drivers/net/null/rte_eth_null.c
--- ./dpdk-2.1.0/drivers/net/null/rte_eth_null.c.null	2015-08-17 13:35:37.000000000 -0400
+++ ./dpdk-2.1.0/drivers/net/null/rte_eth_null.c	2015-08-26 10:34:03.060418807 -0400
@@ -71,8 +71,13 @@ struct pmd_internals {
 	unsigned nb_rx_queues;
 	unsigned nb_tx_queues;
 
-	struct null_queue rx_null_queues[1];
-	struct null_queue tx_null_queues[1];
+	/*
+ 	 * NH these should both be 1, but we want to work
+ 	 * around a gcc compiler error here
+ 	 * as -Werror=array-bounds trips over this when its 1
+ 	 */
+	struct null_queue rx_null_queues[2];
+	struct null_queue tx_null_queues[2];
 };
 
 
