diff -up ./app/test/Makefile.libfix ./app/test/Makefile
--- ./app/test/Makefile.libfix	2017-02-07 11:57:20.198263877 -0500
+++ ./app/test/Makefile	2017-02-07 11:57:34.700156907 -0500
@@ -187,6 +187,7 @@ ifeq ($(CONFIG_RTE_LIBRTE_PMD_NULL),y)
 SRCS-$(CONFIG_RTE_LIBRTE_PMD_BOND) += test_link_bonding_rssconf.c
 ifeq ($(CONFIG_RTE_BUILD_SHARED_LIB),y)
 LDLIBS += -lrte_pmd_null
+LDLIBS += -lrte_pmd_ring
 endif
 endif
 
